channeldb/mp_payment_test.go

```diff
@@ -122,3 +125,152 @@ func TestRegistrable(t *testing.T) {
 		})
 	}
 }
+
+// TestUpdatePaymentState checks that the method updateState updates the
+// MPPaymentState as expected.
+func TestUpdatePaymentState(t *testing.T) {
+	t.Parallel()
+
+	// paymentHash is the identifier on paymentLifecycle.
+	preimage := lntypes.Preimage{}
+	failureReasonError := FailureReasonError
+
+	testCases := []struct {
+		name     string
+		payment  *MPPayment
+		totalAmt int
+
+		expectedState *MPPaymentState
+		errExpected   error
+	}{
+		{
+			// Test that when the sentAmt exceeds totalAmount, the
+			// error is returned.
+			name: "amount exceeded error",
+			// SentAmt returns 90, 10
+			// TerminalInfo returns non-nil, nil
+			// InFlightHTLCs returns 0
+			payment: &MPPayment{
+				HTLCs: []HTLCAttempt{
+					makeSettledAttempt(100, 10, preimage),
+				},
+			},
+			totalAmt:    1,
+			errExpected: ErrSentExceedsTotal,
+		},
+		{
+			// Test that when the htlc is failed, the fee is used.
```

Comment:
Q: I guess this should be: `// Test that when the htlc is failed, the fee is not used.`